datalab/datalab_session/data_operations/median.py

```diff
@@ -40,25 +42,24 @@ def wizard_description():
         }
     
     def operate(self):
-
-        input = self.input_data.get('input_files', [])
-
-        if len(input) <= 1:
-            raise ClientAlertException('Median needs at least 2 files')
-
-        log.info(f'Executing median operation on {len(input)} files')
-
-        image_data_list = self.get_fits_npdata(input)
-
-        cropped_data_list = crop_arrays(image_data_list)
-        stacked_data = np.stack(cropped_data_list, axis=2)
-
-        # using the numpy library's median method
-        median = np.median(stacked_data, axis=2)
+        # Getting/Checking the Input
+        input_list = self.input_data.get('input_files', [])
+        if len(input_list) <= 1: raise ClientAlertException('Median needs at least 2 files')
+        comment = f'Datalab Median on {", ".join([image["basename"] for image in input_list])}'
+        log.info(comment)
+
+        input_FITS_list = []
+        for index, input in enumerate(input_list, start=1):
+            input_FITS_list.append(FITSFileReader(input['basename'], input['source']))
```

Comment:
I'm not 100% sure but there may be a subtle difference in behavior here to what we had before. Before, the fits file was opened in a function and the hdu with science data extracted and returned, presumably "closing" the fits file after that function returned. But now the fits file is opened and hdu extracted and both are stored in the class thats stored in this array until the end of this operation. So I think the result might be higher memory usage? But maybe python doesn't garbage collect it file close anyway so maybe its the same too.

Reply:
good point, I changed the class to not store the hdul now and just open it when needed, that way we don't have a lot of files open in memory.